packages/venia-ui/lib/components/CartPage/cartPage.js

```diff
@@ -33,6 +33,15 @@ const CartPage = props => {
         ) : null;
     }, [classes.sign_in, handleSignIn, isSignedIn]);
 
+    const productListing = hasItems ? (
+        <ProductListing />
+    ) : (
+        <h3>There are no items in your cart.</h3>
+    );
+
+    const priceAdjustments = hasItems ? <PriceAdjustments /> : null;
```

Comment:
Same here. Can we use `useMemo`? At most we will only be caching 2 versions of `priceAdjustments` either the component or null.